### PR TITLE
DashboardScene: Add time picker keybindings

### PR DESCRIPTION
**Pull request by dprokop**
This adds 
- a keybinding (ctrl+z, t z) to allow time range zoom out in dashboard scene.
- a keybinding (t left, tright) for navigating time range back and forth

~Additionally, it adds a zoom function [that's currently implemented in scenes](https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/SceneTimePicker.tsx#L74) to grafana/data `rangeUtil` so that it can be reused.~

It adds `dashboardSceneGraph` utility that will allow us to traverse the specific DashboardScene hierarchy. For now I've just added `getTimePicker` and respective tests.

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts [hunk 1]
```diff
@@ -79,6 +80,20 @@ export function setupKeyboardShortcuts(scene: DashboardScene) {
     onTrigger: () => sceneGraph.getTimeRange(scene).onRefresh(),
   });
 
+  // Zoom out
+  keybindings.addBinding({
+    key: 't z',
+    onTrigger: () => {
+      handleZoomOut(scene);
+    },
+  });
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:89]: We are still missing the t left arrow and t right arrow shortcuts (to move time window left or right)

**Review by torkelo — APPROVED** [review]

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:89]: yup, will add them soon to this PR

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts [hunk 2]
```diff
@@ -128,3 +157,23 @@ export function toggleVizPanelLegend(vizPanel: VizPanel) {
 function hasLegendOptions(optionsWithLegend: unknown): optionsWithLegend is OptionsWithLegend {
   return optionsWithLegend != null && typeof optionsWithLegend === 'object' && 'legend' in optionsWithLegend;
 }
+
+function handleZoomOut(scene: DashboardScene) {
+  const timePicker = dashboardSceneGraph.getTimePicker(scene);
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:162]: @ivanortegaalba  - I'm adding this util to allow abstracting DashboardScene traversal to retrieve certain objects. I think you may find it useful for your General Settings work. 

FYI @torkelo  - think this util will be good to provide a way to make sure that structural changes in DashboardScene won't bring regressions in the future.

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:89]: Those are added now

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx [hunk 3]
```diff
@@ -108,7 +107,8 @@ export function TimeRangePicker(props: TimeRangePickerProps) {
 
   const styles = useStyles2(getStyles);
   const { modalBackdrop } = useStyles2(getModalStyles);
-  const hasAbsolute = isDateTime(value.raw.from) || isDateTime(value.raw.to);
+  const hasAbsolute = !rangeUtil.isRelativeTime(value.raw.from) || !rangeUtil.isRelativeTime(value.raw.to);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx:110]: This change is required, as in the old arch range.raw.to/from could have been expressed as moment object or string. In Scenes we guarantee it's always string.

**Review by torkelo — APPROVED** [review]

**Pull request merged**